### PR TITLE
about ParamPattern and java.util.Date

### PR DESCRIPTION
**Pull request by ScienJus**
from https://github.com/decebals/pippo/issues/256

When pippo start, `DefaultControllerHandler.findMethod`will create some `ParameterValue` and call them `to` method, is there a better ways like call a method named `isSupport`? whatever, when method has a `java.util.Date` parameter, there will a error beacause `ParameterValue.to` is not support `java.util.Date` without `pattern`, and `DefaultControllerHandler.findMethod` do not parse `@ParamPatter`。

So I did these things:
1. when method has a `java.util.Date` parameter and without `@ParamPattern`, it will be converted  from a unix timestamp like `System.currentTimeMillis()`, because it's useful and not be supported by `@ParamPattern`.
2. in `DefaultControllerHandler.findMethod`, find all `@ParamPattern.value` from method parameters. and sent to `ParameterValue.to`

I could not find a good way to test it in unit test, but i create a real web application and it worked fine.

### CODE REVIEW

File: pippo-controller/src/test/java/ControllerHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ */
+import org.junit.Test;
+import ro.pippo.controller.Controller;
+import ro.pippo.controller.DefaultControllerHandler;
+import ro.pippo.core.Param;
+import ro.pippo.core.ParamPattern;
+
+/**
+ * @author ScienJus
+ * @date 2016/3/2.
```
Comment (decebals) [inline pippo-controller/src/test/java/ControllerHandlerTest.java]: Please remove this line.

File: pippo-controller/src/test/java/ControllerHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ */
+public class ControllerHandlerTest {
+
+    @Test
+    public void testFindMethodWithDate() throws Exception {
```
Comment (decebals) [inline pippo-controller/src/test/java/ControllerHandlerTest.java]: It's an idea to split this method in two methods (`testDateParam` and `testDateParamWithPattern`).
What we can do for both tests in the current implementation is to create a controller handler and to use an `assertNotNull(handler.getMethod())`.
I see a supplement assert related  to `parameterNames` if we create in `DefaultControllerHandler` a getter for `parameterNames` property. In this mode we can test the parameter names size and the paramater name.

**Comment by decebals** [comment]
> when method has a java.util.Date parameter and without @ParamPattern, it will be converted from a unix timestamp like System.currentTimeMillis(), because it's useful and not be supported by @ParamPattern.

I don't know if this helps. I am not sure that I can supply you the unix timestamp from the frontend. In this situation maybe a hardcoded default date pattern is a better and simple idea.

@gitblit What you think?

Related to the code fragment from `ParameterValue`:
```java
if (pattern == null) {
    // no defined pattern, use type defaults
    if (type == java.sql.Date.class) {
        return toSqlDate();
    }

    if (type == Time.class) {
        return toSqlTime();
    }

    if (type == Timestamp.class) {
        return toSqlTimestamp();
    }
    // support java.util.Date
    if (Date.class.isAssignableFrom(type)) {
        //default from unix timestamp like System.currentTimeMillis()
        return toDateFromUnixTimestamp();
    }
} else {
    if (Date.class.isAssignableFrom(type)) {
        Date date = toDate(pattern);
        if (type == Date.class) {
            return date;
        } else if (type == java.sql.Date.class) {
            return new java.sql.Date(date.getTime());
        } else if (type == Time.class) {
            return new Time(date.getTime());
        } else if (type == Timestamp.class) {
            return new Timestamp(date.getTime());
        }
    }
}        
```

maybe it's better:
```java
if (Date.class.isAssignableFrom(type)) {
   // make something with 'pattern' if pattern is null // <-- HERE
    Date date = toDate(pattern);
    if (type == Date.class) {
        return date;
    } else if (type == java.sql.Date.class) {
        return new java.sql.Date(date.getTime());
    } else if (type == Time.class) {
        return new Time(date.getTime());
    } else if (type == Timestamp.class) {
        return new Timestamp(date.getTime());
    }
}
```

**Comment by ScienJus** [comment]
ok, also people can modify the default date format is better.

**Pull request closed without merging**